doc/api/process.markdown

```diff
@@ -16,6 +16,34 @@ continue.
 as [`process.exit()`][] or uncaught exceptions, and should not be used as an
 alternative to the `'exit'` event unless the intention is to schedule more work.
 
+## Event: 'deprecation'
```

Comment:
I split this out specifically because @dougwilson pointed out that `depd` already makes use of `process.on('deprecation')`. The idea is to ensure more alignment between core and what's happening in userland.